tuning/data/setup_dataprocessor.py

```diff
@@ -68,6 +91,17 @@ def _process_dataconfig_file(
         tokenizer=tokenizer,
         additional_data_handlers=additional_data_handlers,
     )
+    if processor.processor_config.streaming:
+        if train_args.max_steps < 1:
+            logging.error(
+                "ValueError: `--max_steps` must be set when streaming is set in data \
+                            preprocessor config"
+            )
+            raise ValueError(
+                "`--max_steps` must be set when streaming is set in data preprocessor config"
+            )
+        if train_args.num_train_epochs:
+            logging.warning("`--num_train_epochs` will be overwritten by `--max_steps`")
```

Comment:
Since, this is the default behaviour of the HF transformers - https://github.com/huggingface/transformers/blob/ec7afad60909dd97d998c1f14681812d69a15728/src/transformers/trainer.py#L695 

We can avoid this warning all together.